src/crawlers/cheerio_crawler.js

```diff
@@ -354,14 +384,22 @@ class CheerioCrawler {
      *
      * @ignore
      */
-    async _handleRequestFunction({ request, autoscaledPool }) {
+    async _handleRequestFunction({ request, autoscaledPool, session }) {
         if (this.prepareRequestFunction) await this.prepareRequestFunction({ request });
         const { dom, isXml, body, contentType, response } = await addTimeoutToPromise(
-            this._requestFunction({ request }),
+            this._requestFunction({ request, session }),
             this.requestTimeoutMillis,
             `CheerioCrawler: request timed out after ${this.requestTimeoutMillis / 1000} seconds.`,
         );
 
+        if (this.useSessionPool) {
+            this._handleBlockedRequest(session, response.statusCode);
+        }
+
+        if (this.persistCookiesPerSession) {
+            session.setCookiesToJar(response);
```

Comment:
This is not very intuitive. When calling `setCookies...` I'd expect to pass `cookies`, not `response`.

`setResponseCookiesToJar` `setCookiesFromResponseToJar` ? Feels weird. I'd probably prefer `setResponse` or `putResponse`. The function can then do more than setting cookies.